service/storage_service.cc

```diff
@@ -6023,6 +6023,14 @@ future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables(
         });
     }
 
+    // Retrieval of tablet stats must be serialized with mutation to token metadata,
+    // as the former requires tablet id stability.
+    // If tablet split is finalized while retrieving stats, the saved erm, used by all
+    // shards, can have a lower tablet count than the one in a particular shard, causing
+    // an abort as tablet map requires that any id feeded into it is lower than its
+    // current tablet count.
+    auto tmlock = co_await get_token_metadata_lock();
+
```

Comment:
btw, the check below:

```
s < locator::tablet_transition_stage::cleanup
```

is no longer accurate because stages are non-linear, we have revert_migration and end_migration. Better to have a helper function with a switch for all stages.

Reply:
Will send another patch to fix this. Thanks.